### ISSUE TITLE
Prettier breaks file when wrapping variables in curly braces [WebStorm]

### ISSUE THREAD

**Issue by laszlo1337** [sveltejs/prettier-plugin-svelte#405]
**Prettier 3.0.3**

In WebStorm, when editing .svelte files, when trying to wrap text in {} by selecting the text and pressing shift + {
prettier breaks completely. Steps:

1. Create example component
![image](https://github.com/prettier/prettier/assets/24696743/bd974fdc-3985-4f8f-9be7-f0ed9bfecce8)

2. Select text that needs to be wrapped in {}
![image](https://github.com/prettier/prettier/assets/24696743/2c24d2b0-4d14-4d76-b4f0-21e79510079e)

3. Press `Shift + {` and experience the file getting screwed
![image](https://github.com/prettier/prettier/assets/24696743/9d0770de-0ba8-40b0-9b8e-6e355b2b1a6c)

This does not happen in .js files, and wrapping in square brackets works as intended (`Shift + [`)

`prettierrc` file 
```
{
  "useTabs": true,
  "singleQuote": true,
  "trailingComma": "all",
  "printWidth": 180,
  "semi": false,
  "plugins": [
    "prettier-plugin-svelte"
  ],
  "pluginSearchDirs": [
    "."
  ],
  "overrides": [
    {
      "files": "*.svelte",
      "options": {
        "parser": "svelte"
      }
    }
  ]
}

```
This happens also when committing with "Reformat code" and "Optimize imports" checkboxes checked. The code was committed in a broken state.

**Expected behavior:**
Should format the component without breaking

**Comment by dummdidumm** [comment]
Closing as duplicate of https://github.com/sveltejs/prettier-plugin-svelte/issues/233 and https://github.com/sveltejs/prettier-plugin-svelte/issues/242
TLDR: prettier does not support plugin like prettier-plugin-svelte to partake in the "format range" option properly, and so when your editor does auto-format on code action, it will likely invoke prettier with the range option, resulting in the given gibberish.
